src/hidimstat/knockoffs.py

```diff
@@ -178,268 +108,261 @@ def model_x_knockoff(
-    desired FDR level. It then aggregates the p-values using the quantile aggregation
-    procedure and identifies the selected variables based on the aggregated p-values.
-    """
-    n_bootstraps = len(test_scores)
-    pvals = np.array(
-        [_empirical_knockoff_pval(test_scores[i]) for i in range(n_bootstraps)]
-    )
-
-    aggregated_pval = quantile_aggregation(
-        pvals, gamma=gamma, adaptive=adaptive_aggregation
-    )
-
-    threshold = fdr_threshold(
-        aggregated_pval,
-        fdr=fdr,
-        method=fdr_control,
-        reshaping_function=reshaping_function,
-    )
-    selected = np.where(aggregated_pval <= threshold)[0]
-
-    return selected, aggregated_pval, pvals
-
-
-def _stat_coefficient_diff(X, X_tilde, y, estimator, fdr, preconfigure_estimator=None):
-    """
-    Compute the Lasso Coefficient-Difference (LCD) statistic by comparing original and knockoff coefficients.
-
-    This function fits a model on the concatenated original and knockoff features, then
-    calculates test statistics based on the difference between coefficient magnitudes.
-
-    Parameters
-    ----------
-    X : ndarray of shape (n_samples, n_features)
-        Original feature matrix.
-
-    X_tilde : ndarray of shape (n_samples, n_features)
-        Knockoff feature matrix.
-
-    y : ndarray of shape (n_samples,)
-        Target values.
-
-    estimator : estimator object
-        Scikit-learn estimator with fit() method and coef_ attribute.
-        Common choices include LassoCV, LogisticRegressionCV.
-
-    fdr : float
-        Target false discovery rate level between 0 and 1.
-
-    preconfigure_estimator : callable, default=None
-        Optional function to configure estimator parameters before fitting.
-        Called with arguments (estimator, X, X_tilde, y).
-
-    Returns
-    -------
-    test_score : ndarray of shape (n_features,)
-        Feature importance scores computed as |beta_j| - |beta_j'|
-        where beta_j and beta_j' are original and knockoff coefficients.
-
-    ko_thr : float
-        Knockoff threshold value used for feature selection.
-
-    selected : ndarray
-        Indices of features with test_score >= ko_thr.
-
-    Notes
-    -----
-    The test statistic follows Equation 1.7 in Barber & Candès (2015) and
-    Equation 3.6 in Candès et al. (2018).
-    """
-    n_samples, n_features = X.shape
-    X_ko = np.column_stack([X, X_tilde])
-    if preconfigure_estimator is not None:
-        estimator = preconfigure_estimator(estimator, X, X_tilde, y)
-    estimator.fit(X_ko, y)
-    if hasattr(estimator, "coef_"):
-        coef = np.ravel(estimator.coef_)
-    elif hasattr(estimator, "best_estimator_") and hasattr(
-        estimator.best_estimator_, "coef_"
+    def __init__(
+        self,
+        generator=GaussianKnockoffs(cov_estimator=LedoitWolf(assume_centered=True)),
+        statistical_test=lasso_statistic_with_sampling,
+        n_repeat=1,
```

Comment:
In the example we often use `n_repeat=n_bootstrap` 
If n_bootstrap is an accurate name, could we rename the argument as the term bootstrap is commonly used in other libraries. WDYT @bthirion ?